shared/react-native/ios/KeybaseShare/ShareViewController.m

```diff
@@ -0,0 +1,301 @@
+//
+//  ShareViewController.m
+//  KeybaseShare
+//
+//  Created by Michael Maxim on 8/31/18.
+//  Copyright © 2018 Keybase. All rights reserved.
+//
+
+#import "ShareViewController.h"
+#import "keybase/keybase.h"
+#import "Pusher.h"
+#import <MobileCoreServices/MobileCoreServices.h>
+#import <AVFoundation/AVFoundation.h>
+#import "Fs.h"
+
+#if TARGET_OS_SIMULATOR
+const BOOL isSimulator = YES;
+#else
+const BOOL isSimulator = NO;
+#endif
+
+
+@interface ShareViewController ()
+@property NSDictionary* convTarget; // the conversation we will be sharing into
+@property BOOL hasInited; // whether or not init has succeeded yet
+@end
+
+@implementation ShareViewController
+
+- (BOOL)isContentValid {
+    return self.hasInited;
+}
+
+// presentationAnimationDidFinish is called after the screen has rendered, and is the recommended place for loading data.
+- (void)presentationAnimationDidFinish {
+  BOOL skipLogFile = NO;
+  NSError* error = nil;
+  NSDictionary* fsPaths = [[FsHelper alloc] setupFs:skipLogFile setupSharedHome:NO];
+  KeybaseExtensionInit(fsPaths[@"home"], fsPaths[@"sharedHome"], fsPaths[@"logFile"], @"prod", isSimulator, NULL, NULL, &error);
+  if (error != nil) {
+    // If Init failed, then let's throw up our error screen.
+    NSLog(@"Failed to init: %@", error);
+    InitFailedViewController* initFailed = [InitFailedViewController alloc];
+    [initFailed setDelegate:self];
+    [self pushConfigurationViewController:initFailed];
+    return;
+  }
+  [self setHasInited:YES]; // Init is complete, we can use this to take down spinner on convo choice row
+  [self validateContent];
+  
+  NSString* jsonSavedConv = KeybaseExtensionGetSavedConv(); // result is in JSON format
+  if ([jsonSavedConv length] > 0) {
+    NSData* data = [jsonSavedConv dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary* conv = [NSJSONSerialization JSONObjectWithData:data options: NSJSONReadingMutableContainers error: &error];
+    if (!conv) {
+      NSLog(@"failed to parse saved conv: %@", error);
+    } else {
+      // Success reading a saved convo, set it and reload the items to it.
+      [self setConvTarget:conv];
+    }
+  }
+  [self reloadConfigurationItems];
+}
+
+-(void)initFailedClosed {
+  // Just bail out of the extension if init failed
+  [self cancel];
+}
+
+- (NSItemProvider*)firstSatisfiesTypeIdentifierCond:(NSArray*)attachments cond:(BOOL (^)(NSItemProvider*))cond {
+  for (NSItemProvider* a in attachments) {
+    if (cond(a)) {
+      return a;
+    }
+  }
+  return nil;
+}
+
+- (NSMutableArray*)allSatisfiesTypeIdentifierCond:(NSArray*)attachments cond:(BOOL (^)(NSItemProvider*))cond {
+  NSMutableArray* res = [NSMutableArray array];
+  for (NSItemProvider* a in attachments) {
+    if (cond(a)) {
+      [res addObject:a];
+    }
+  }
+  return res;
+}
+
+- (BOOL)isRealURL:(NSItemProvider*)item {
+  // "file URLs" also have type "url", but we want to treat them as files, not text.
```

Comment:
can this be `isTextURL` or something i think it's less confusing

Reply:
How about `isWebURL`?

Reply:
sounds good!